lib/router.js

```diff
@@ -138,7 +138,9 @@ Router.prototype.applyRoutes = function (server, prefix) {
   this.routers.forEach(function (r) {
     var router = r.router;
     var path = prefix + r.path;
-    router.use(self.commonHandlers);
+    // Prepend commonHandlers before nested router's handlers
+    router.commonHandlers = self.commonHandlers.concat(router.commonHandlers);
+    router.use(router.commonHandlers);
```

Comment:
I believe you don't need the second line.  
For example:

```
self.commonHandlers = [fn1, fn2, fn3]
router.commonHandlers = [fn4, fn5]
// then
// with your two lines
router.commonHandlers = self.commonHandlers.concat(router.commonHandlers);
// now router.commonHandlers = [fn1, fn2, fn3, fn4, fn5]  as we want
router.use(router.commonHandlers);
// now router.commonHandlers = [fn1, fn2, fn3, fn4, fn5, fn1, fn2, fn3, fn4, fn5]
```
The reason your test passes is because your `first` middleware does an assignment to [1].
So even though the middleware list has been duplicated, the array gets reset to [1] and then appended to.